src/engine/init.c

```diff
@@ -283,9 +283,12 @@ dss_tgt_nr_get(unsigned int ncores, unsigned int nr, bool oversubscribe)
 		goto retry;
 	}
 
-	if (tgt_nr != nr)
+	if (tgt_nr != nr) {
 		D_PRINT("%d target XS(xstream) requested (#cores %d); "
 			"use (%d) target XS\n", nr, ncores, tgt_nr);
+		if (!oversubscribe)
+			return -DER_INVAL;
```

Comment:
The console message above and the comments of 'oversubscribe' in this function needs be revised.

Reply:
the comments in this function should be revised as well.